File: client-data/src/main/java/com/jraska/github/client/users/data/GitHubUsersApi.java
package com.jraska.github.client.users.data;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

interface GitHubUsersApi {
  @GET("/users") Single<List<GitHubUser>> getUsers(@Query("since") int since);

  @GET("/repos/{owner}/{name}")
  Single<GitHubRepo> getRepo(@Path("owner") String path, @Path("name") String name);
}

File: client-data/src/main/java/com/jraska/github/client/users/data/GitHubUsersApi.kt
package com.jraska.github.client.users.data

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

internal interface GitHubUsersApi {
  @GET("/users")
  fun getUsers(@Query("since") since: Int): Single<List<GitHubUser>>

  @GET("/repos/{owner}/{name}")
  fun getRepo(@Path("owner") path: String, @Path("name") name: String): Single<GitHubRepo>
}
